### DISCUSSION TITLE
Requirements analysis

### DISCUSSION THREAD

**Discussion by smagnaniuni** [progetto-oop-22-23/OOP22-HOMER#4] [General]
Need to clarify:
- What can the user see and do? (Add/remove/edit devices, see graphs, view status, view consumption)
- How will the emulation work? Real time, 24h cycle? Can time be controlled?
- ~~What data needs to be logged? (The graphs will require a timestamp and the corresponding value)~~ design prob?
- Clarify each term (eg. "smart device")
- Give a basic description of what each device will do
  - Do lights need to be dimmable?
  - How will the controller regulate the devices according to temperature/air quality? eg: 
    - When the temperature falls below a certain temp, the A/C will heat the house
    - When the temperature is above a certain pump, the A/C will cool the house
    - When the air quality is below a certain threshold an alert will be displayed/windows will be opened?
  - Devices:
    - Lights
    - Outlets
    - Locks
    - Windows
    - Rolling Blinds
    - A/C
    - Temperature Sensor
    - Air Quality Sensor

**Comment by michelececcacci** [comment]
> How will the emulation work? Real time, 24h cycle? Can time be controlled?

What do you mean by real time? fetching the values from external APIs? I thought we could also randomize the values(inside a reasonable range). Wouldn't be a problem with good unit testing, and we could have many different situations.

**Reply by smagnaniuni** [reply]
Not the actual real time, but more like whether it's a live simulation or the time is "static" (think we decided for it to run continuosly, but we need to specify it, and more importantly we need to specify whether it's required to have a knowledge of days or not)

**Reply by michelececcacci** [reply]
this makes more sense than what i wrote previously. I feel like we could start with "static" time, and maybe add an option to have it "dynamic" later

**Comment by michelececcacci** [comment]
Lights: I think dimmable makes sense
Outlets: on/off
Locks: on/off
Windows: "dimmable" (they can be opened at different angles)
Rolling Blinds: same as windows
A/C: dimmable
Temperature Sensor: on/off
Air Quality Sensor: on/off, with some settings (we could also have a curve for dealing with temperatures)

**Comment by michelececcacci** [comment]
>What can the user see and do? (Add/remove/edit devices, see graphs, view status, view consumption)

They should be able to add/remove devices, view status and consumption. 
We also need graphs, but i think it will be one of the last things to be implemented in the project. (chronologically speaking)
